web/packages/teleport/src/UnifiedResources/Resources.tsx

```diff
@@ -278,9 +278,14 @@ export function Resources() {
           <ErrorBoxInternal>
             <Danger>
               {attempt.statusText}
-              <Box flex="0 0 auto" ml={2}>
-                <ButtonLink onClick={onRetryClicked}>Retry</ButtonLink>
-              </Box>
+              {/* we don't want them to try another request with failed predicate expression, it will just fail again. */}
+              {!attempt.statusText.includes(
+                'failed to parse predicate expression'
```

Comment:
This will break as soon as the error message changes.
To avoid this, I would put a comment in the Go code where that message is defined, warning devs not to change it.

Reply:
Added a comment and added it to the test. which really feels backwards like I'm super hardening a fickle conditional anyway but, we can refactor the frontend `useAttempt` to return a status code that we can work with instead in a future PR. I added a TODO

Reply:
I agree. it returns a 400 currently. we just need to refactor the `useAttempt` hook and its callsites to include this statusCode on error which might blow up the PR a bit (all of web uses it)

Reply:
Actually, this refactor might not be too big. I think I can squeeze it into this PR. hold off on further review

Reply:
Thanks for pushing on this. A similar refactor happened with the async code that Connect uses `useAsync` and I remember it being a large thing. For `useAttempt` it turned out to be easy since all the request stuff is updated manually by the caller. I added an optional `statusCode` field and didn't have to update any of the other callsites (although I'm sure I can find some that would benefit from this change. Thats for another day).